File: app/src/main/java/com/anglersparadise/ui/tank/FishTankActivity.java
// app/src/main/java/com/anglersparadise/ui/tank/FishTankActivity.java
package com.anglersparadise.ui.tank;

import android.os.Bundle;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.anglersparadise.databinding.ActivityFishTankBinding;

public class FishTankActivity extends AppCompatActivity {

    private ActivityFishTankBinding binding;
    private TankViewModel vm;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityFishTankBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        vm = new ViewModelProvider(this).get(TankViewModel.class);

        binding.btnOpenBasket.setOnClickListener(v ->
                new FishBasketDialog().show(getSupportFragmentManager(), "basket")
        );
        binding.btnBackToLake.setOnClickListener(v -> finish());
        binding.btnReleaseAll.setOnClickListener(v ->
                new AlertDialog.Builder(this)
                        .setTitle("Empty tank?")
                        .setMessage("Are you sure you want to release all fish?")
                        .setPositiveButton("Yes", (d, w) -> vm.releaseAll())
                        .setNegativeButton("Cancel", null)
                        .show()
        );

        vm.tankFish.observe(this, list -> {
            if (list == null) return;
            binding.tankCanvas.setFish(list);
            binding.count.setText("Fish in tank: " + list.size());
        });
    }
}

File: app/src/main/java/com/anglersparadise/ui/tank/FishTankActivity.kt
// app/src/main/java/com/anglersparadise/ui/tank/FishTankActivity.kt

package com.anglersparadise.ui.tank

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.lifecycleScope
import androidx.lifecycle.repeatOnLifecycle
import com.anglersparadise.databinding.ActivityFishTankBinding
import kotlinx.coroutines.launch

class FishTankActivity : AppCompatActivity() {

    private lateinit var binding: ActivityFishTankBinding
    private val vm: TankViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityFishTankBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnOpenBasket.setOnClickListener {
            FishBasketDialog().show(supportFragmentManager, "basket")
        }

        binding.btnBackToLake.setOnClickListener { finish() }

        binding.btnReleaseAll.setOnClickListener {
            AlertDialog.Builder(this)
                .setTitle("Empty tank?")
                .setMessage("Are you sure you want to release all fish?")
                .setPositiveButton("Yes") { _, _ -> vm.releaseAll() }
                .setNegativeButton("Cancel", null)
                .show()
        }

        // Observe tank fish -> feed the canvas + list text
        lifecycleScope.launch {
            repeatOnLifecycle(Lifecycle.State.STARTED) {
                vm.tankFish.collect { list ->
                    binding.tankCanvas.setFish(list)
                    binding.count.text = "Fish in tank: ${list.size}"
                }
            }
        }
    }
}
